caravel/controllers/listings.py

```diff
@@ -18,6 +18,14 @@
 def search_listings():
     """Display a list of listings that match the given query."""
 
+    # Fix session handler if not initialized
+    if not session.get("view"):
```

Comment:
Nah, I think we should use ?key=value — just intercepting it and forwarding it as we do with ?q.